minidump-processor/src/op_analysis.rs

```diff
@@ -338,6 +348,18 @@ mod amd64 {
                     | Opcode::XSETBV
             )
         }
+
+        /// Since we only detect inconsistencies in non-canonical crashes if we can derive all its access,
+        /// this function always return false for opcodes that are not `AccessDerivableOpcode`
+        fn is_only_gpf_when_non_canonical(instruction: Instruction) -> bool {
+            let Some(opcode) = AccessDerivableOpcode::from_opcode(instruction.opcode()) else {
+                return false;
+            };
+            !matches!(
+                opcode,
+                AccessDerivableOpcode::MOVAPS | AccessDerivableOpcode::UCOMISS
```

Comment:
`MOVAPS` description in Intel manual states that it raises a general-protection fault when the address is unaligned, but `UCOMISS` instruction doesn't mention any alignment requirements, nor that it raise #GP for other conditions. Do we have crashes where the crashing instruction is `UCOMISS` and we get a #GP without a non-canonical address? IIRC we had some odd crashes involving `UCOMISS` but they were cause by a bug in QEMU, not by actual CPU behavior.

Reply:
Ah I was under the impression that `UCOMISS` has alignment requirements. Should have looked at the Intel manual more closely. And no I did not see any #GP crashes with `UCOMISS` without non-canonical address.
The QEMU bug causes an actual access violation instead of a GP, since it's accessing some extra bytes that it's not supposed to.

Reply:
@gabrielesvelto Is the QEMU bug the reason why you suggested in https://github.com/rust-minidump/rust-minidump/issues/732#issuecomment-2373323897 to drop non-canonical inconsistency detection for `UCOMISS`? If so it's probably fine to keep `UCOMISS` and only drop the detection for `MOVAPS`?

Reply:
Possibly, it was a mistake on my part.

Reply:
Ah ok, no worries. I'll change it to only dropping `MOVAPS`.